src/main/java/org/jboss/threads/JBossExecutors.java

```diff
@@ -350,7 +355,9 @@ static ClassLoader getAndSetContextClassLoader(final Thread thread, final ClassL
      * @param classLoader the new context class loader
      */
     static void setContextClassLoader(final Thread thread, final ClassLoader classLoader) {
-        unsafe.putObject(thread, contextClassLoaderOffs, classLoader);
+        if (unsafe.getObject(thread, contextClassLoaderOffs) != classLoader) {
+            unsafe.putObject(thread, contextClassLoaderOffs, classLoader);
```

Comment:
Too bad there's no `compareAndSetIfDifferent` :)

Do we need to care about the potential race introduced by uncoupled get and set?

Reply:
I hope not, given that's a thread class loader, I hope no others would modify it concurrently, but I'm not an expert on this subject

Reply:
I think the proposed patch has the same semantics as the existing one, in terms of races